test/controllers/CyaCreateRecordControllerSpec.scala

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import base.SpecBase
+import models.{Commodity, UserAnswers}
+import org.scalatestplus.mockito.MockitoSugar
+import pages._
+import play.api.test.FakeRequest
+import play.api.test.Helpers._
+import uk.gov.hmrc.play.bootstrap.binders.RedirectUrl
+import viewmodels.govuk.SummaryListFluency
+import views.html.CyaCreateRecordView
+
+class CyaCreateRecordControllerSpec extends SpecBase with SummaryListFluency with MockitoSugar {
+
+  "CyaNirmsNiphlController" - {
+
+    "for a GET" - {
+
+      "must return OK and the correct view with valid mandatory data" in {
+
+        val answers =
+          UserAnswers(userAnswersId)
+            .set(TraderReferencePage, "123")
+            .success
+            .value
+            .set(CommodityCodePage, Commodity("654321", "Description"))
+            .success
+            .value
+            .set(CountryOfOriginPage, "1")
+            .success
+            .value
+            .set(HasGoodsDescriptionPage, false)
+            .success
+            .value
+
+        val application = applicationBuilder(userAnswers = Some(answers)).build()
+
+        running(application) {
+          val request = FakeRequest(GET, routes.CyaCreateRecordController.onPageLoad.url)
+
+          val result = route(application, request).value
+
+          val view = application.injector.instanceOf[CyaCreateRecordView]
+          val list = SummaryListViewModel(
+            rows = Seq.empty
+          )
+
+          status(result) mustEqual OK
+          contentAsString(result) mustEqual view(list)(request, messages(application)).toString
+        }
+      }
+
+      "must return OK and the correct view with all data (including optional)" in {
+
+        val userAnswers = UserAnswers(userAnswersId)
```

Comment:
Generating userAnswers can be extracted to a function, if we need it in a lot of specs it might be good to create a Generator class and link it to the SpecBase, so we can do something like:


`def userAnswers(traderReference, commodityCode, countryOfOrigin...)`

Reply:
I've thought a lot about this and having been faffing around with it on the branch. I think it makes it clearer to just set the useranswers in the test as each test needs slightly different useranswers. There is benefit in having empty user answers and maybe mandatory and mandatory + optional, but the others are specific test cases, so it's probably better just to have them in the test

Reply:
I've changed them in both CYA specs, take a look